### ISSUE TITLE
Loading Images w/o target but with "isFromMemoryCache" parameter

### ISSUE THREAD

**Issue by Unihilator** [kean/Nuke#116]
Hey @kean . Need to get isFromMemoryCache status of loaded image. But dont want to use `target`. Is there any way to do this? Was doing this before in Nuke3. 

Thanks)

**Comment by kean** [comment]
Hi! 

Just use anything as a target and use `loadImage(with:into:hander:)` method in which target acts like a cancellation token:

```swift
let viewController =  UIViewController()
loadImage(with: request, into: viewController) { result, isFromMemCache in
    // handle result
}
        
// Use target to canel if necessary
cancelRequest(for: viewController)
```

This is one of my primary use cases, I also use it to [load thumbnail first, fullsize image later](https://github.com/kean/Nuke/blob/master/Documentation/Guides/Tips%20and%20Tricks.md#show-a-low-res-image-first-and-swap-to-a-higher-res-one-when-it-arrives).

I'll think what I can do in Nuke 5.1 to add something like `loadImage(with:completion:` to `Manager` class.

**Comment by kean** [comment]
I've realized that this kind of API is actually necessary for Preheater. I'll do some damage control in Nuke 5.0.1 to address this issue. Stay tuned.

**Comment by Unihilator** [comment]
Great man) thanks

**Comment by kean** [comment]
New APIs available in [Nuke 5.0.1](https://github.com/kean/Nuke/releases/tag/5.0.1):

```swift
class Manager {
    public func loadImage(with url: URL, token: CancellationToken?, completion: @escaping (Result<Image>) -> Void)
    public func loadImage(with request: Request, token: CancellationToken?, completion: @escaping (Result<Image>) -> Void)
}

They both check memory cache asynchronously and they both call completion closure asynchronously on the main thread. Instead of `isFromMemoryCache` just use memory cache directly:

```swift
let image = Cache.shared[request]
```
